qemu/tests/sr_iov_hotplug.py

```diff
@@ -33,14 +32,26 @@ def run(test, params, env):
     :param env:    Dictionary with test environment.
     """
 
+    def check_interface(iface, nic_filter):
+        cmd = "ifconfig %s" % str(iface)
+        status, output = session.cmd_status_output(cmd)
+        if status:
+            test.error("Guest command '%s' fail with output: %s." % (cmd, output))
+        if re.findall(nic_filter, output, re.MULTILINE | re.DOTALL):
+            return True
+        return False
+
     def get_active_network_device(session, nic_filter):
         devnames = []
         cmd = "ifconfig -a"
+        nic_reg = "\w+(?=: flags)|\w+(?=\s*Link)"
         status, output = session.cmd_status_output(cmd)
         if status:
-            msg = "Guest command '%s' fail with output: %s." % (cmd, output)
-            raise error.TestError(msg)
-        devnames = re.findall(nic_filter, output)
+            test.error("Guest command '%s' fail with output: %s." % (cmd, output))
+        ifaces = re.findall(nic_reg, output)
+        for iface in ifaces:
+            if check_interface(str(iface), nic_filter):
```

Comment:
Sure, so when we have multiple interfaces in guest after sriov hotplug, we iterate each interface and check the active interfaces with ip and mac.